onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -121,4 +127,303 @@ std::shared_ptr<KernelRegistry> MKLDNNExecutionProvider::GetKernelRegistry() con
   static std::shared_ptr<KernelRegistry> kernel_registry = onnxruntime::mkl_dnn::GetMklDnnKernelRegistry();
   return kernel_registry;
 }
+
+bool MKLDNNExecutionProvider::UseSubgraph(const onnxruntime::GraphViewer& graph_viewer,
+                                          const std::vector<const KernelRegistry*>& kernel_registries,
+                                          std::vector<std::unique_ptr<ComputeCapability>>& result) const {
+  // switch between mkldnn-vanilla and mkldnn-subgraph implementation using
+  // MKLDNN_SUBGRAPH environment variable
+  bool use_subgraph = true;
+
+  const char* env = getenv("ORT_MKLDNN_SUBGRAPH");
+  int use_subgraph_env = 0;
+  if (env != nullptr)
+    use_subgraph_env = atoi(env);
+
+  if (use_subgraph_env == 0) {
```

Comment:
the logic here is a little bit confusing because you have two variables use_subgraph and use_subgraph_env
use_subgraph defaults to true, yet the default behavior is false (becuase use_subgraph_env is default to 0)
is it possible to simplify it?

Also, if we are confident that we can run all models now, maybe we should default to true

Reply:
Yes. I will change it to 

```
  bool use_subgraph = true;
  const char* env = getenv("ORT_MKLDNN_SUBGRAPH");
  if (env != nullptr) {
    if (atoi(env) == 0) {
      use_subgraph = false;
      result = IExecutionProvider::GetCapability(graph_viewer, kernel_registries);
    }
  }
  return use_subgraph;
```